plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/mapping/IdentifierMappingModule.java

```diff
@@ -57,6 +58,7 @@ protected void setup(Binder binder)
         binder.bind(DefaultIdentifierMapping.class).in(SINGLETON);
 
         MappingConfig config = buildConfigObject(MappingConfig.class);
+        newOptionalBinder(binder, CachingIdentifierMapping.class);
```

Comment:
Do the changes in this class deserve a comment. It's not obvious to me why this change was needed (at least until now).

Reply:
Added.

Reply:
Thanks. This however feels a bit odd to do.

@kokosing Do we prefer this or should we add an `invalidate` method to `IdentifierMapping` so that we don't need optional binding to a concrete impl (most impls. will implement a no-op `invalidate` method)?

Reply:
+1 on @hashhar's suggestion, it is weird to provide both an optional binding and a non-optional binding

Reply:
Its fully allong `OptionalBinder` docs. 
Also this variant is provided as a first option choice.
https://google.github.io/guice/api-docs/4.2/javadoc/com/google/inject/multibindings/OptionalBinder.html

I know it looks weird at first glimps but i found it preatty clear after a while.

We bind 
- optional to some interface A
- then we bind impl. B to interface A

Optional<A> -> A -> B